Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedListItemView.swift

```diff
@@ -0,0 +1,84 @@
+import ArcGIS
+
+public struct PreplannedListItemView: View {
+    /// The view model for the preplanned map.
+    @ObservedObject var preplannedMapModel: PreplannedMapModel
+    /// A Boolean value indicating whether the preplanned map area can be downloaded.
+    @State private var canDownload: Bool?
+    
+    public var body: some View {
+        HStack {
+            HStack {
+                let preplannedMapArea = preplannedMapModel.preplannedMapArea
+                if let thumbnail = preplannedMapModel.preplannedMapArea.portalItem.thumbnail {
+                    LoadableImageView(loadableImage: thumbnail)
+                        .frame(width: 64, height: 44)
+                }
+                VStack(alignment: .leading, spacing: 2) {
+                    Text(preplannedMapArea.portalItem.title)
+                        .font(.headline)
+                    if !preplannedMapArea.portalItem.description.isEmpty {
+                        Text(preplannedMapArea.portalItem.description)
+                            .font(.caption)
+                            .foregroundColor(.secondary)
+                            .lineLimit(2)
+                    }
+                }
+                Spacer()
+                switch preplannedMapModel.result {
+                case .success:
+                    Image(systemName: "checkmark.circle.fill")
+                case .failure:
+                    Image(systemName: "exclamationmark.circle")
+                        .foregroundColor(.red)
+                case .none:
+                    if let canDownload {
+                        if !canDownload {
+                            // Map is still packaging.
+                            Image(systemName: "clock.badge.xmark")
+                        } else {
+                            // Map package is available for download.
+                            Image(systemName: "arrow.down.circle")
+                        }
+                    } else {
+                        ProgressView()
+                            .frame(maxWidth: .infinity)
+                    }
+                }
+            }
+            .onReceive(preplannedMapModel.preplannedMapArea.$loadStatus) { status in
+                let packagingStatus = preplannedMapModel.preplannedMapArea.packagingStatus
+                if status == .failed || packagingStatus == .processing {
+                    // If the preplanned map area fails to load, it may not be packaged.
+                    canDownload = false
+                } else if packagingStatus == .complete || packagingStatus == nil {
+                    // Otherwise, check the packaging status to determine if the map area is
+                    // available to download.
+                    canDownload = true
+                }
+            }
+            .task {
+                do {
+                    try await preplannedMapModel.preplannedMapArea.load()
+                } catch {
+                    print(error)
```

Comment:
I think the same red exclamation mark can be used for all the errors related to a single `preplannedMapModel`. Since we need to handle the error incrementally, I'd suggest just display the error description under the caption for now.

I'm open to where you want to hold the error. I think holding it in the data model makes more sense. As we move forward, there will be more errors from various methods and instead of scattering errors around different views, having them altogether in the model makes it easy to adjust the error handling logics.